src/zope/server/buffers.py

```diff
@@ -98,11 +98,15 @@ def prune(self):
             if not data:
                 break
             nf.write(data)
+        self.file.close()
```

Comment:
I'm confused by the local assignment of `file = self.file`. Is there a chance `self.file` could change out from under us while this method is running, in which case `file` would leak? Or is this just meant to be a performance optimization? Could the whole method body be wrapped in `with self.file as file`?

Reply:
Huh, I didn't even notice.

My gut feeling says `with self.file` would be wrong: it would close the file too often.

Reply:
Ah, I missed the premature `return`.